app/src/main/java/com/xa/xpensauditor/SetLimitActivity.java

```diff
@@ -0,0 +1,286 @@
+package com.xa.xpensauditor;
+
+import static com.xa.xpensauditor.AddTransactionActivity.printLog;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import java.lang.ref.Reference;
+import java.util.Calendar;
+
+import android.app.Activity;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.ProgressBar;
+import android.widget.TextView;
+import android.widget.DatePicker;
+import android.util.Log;
+import android.widget.Toast;
+
+
+import com.firebase.client.Firebase;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.auth.FirebaseUser;
+
+import org.w3c.dom.Text;
+
+import java.util.Calendar;
+import java.util.Objects;
+import java.util.List;
+import java.util.ArrayList;
```

Comment:
⚠️ **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.CustomImportOrderCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Wrong lexicographical order for 'java.util.ArrayList' import. Should be before 'org.w3c.dom.Text'.